## This script is used to visualize the input class graph

In [4]:
import pandas as pd

In [5]:
topic_nodes = pd.read_csv("./data/wikipedia_topics_nowikidata.csv")

In [6]:
topic_nodes.head()

category  \
0        Category:Main_topic_classifications   
1        Category:Main_topic_classifications   
2              Category:Academic disciplines   
3  Category:Subfields by academic discipline   
4              Category:Scholars by subfield   

                                     subcategory  
0                   Category:Main topic articles  
1                  Category:Academic disciplines  
2      Category:Subfields by academic discipline  
3                  Category:Scholars by subfield  
4  Category:Anthropologists by field of research

In [7]:
topic_nodes.shape

(339250, 2)

The input data set is a connected graph consisting 339250 edges. Every node of this graph represents a certain cathegory. The graph is not a tree and it can even contain loops.

In [8]:
import networkx as nx

topic_graph = nx.from_pandas_edgelist(topic_nodes, source="category", target="subcategory")
topic_graph.number_of_nodes()

225765

The graph consists of 225765 nodes. Each one represents one category.

In [11]:
import matplotlib as plt

pos=nx.spring_layout(topic_graph)
nx.draw(topic_graph, pos, node_color='#A0CBE2', edge_color='#BB0000', width=2, edge_cmap=plt.cm.Blues, with_labels=True)

plt.savefig("graph.pdf")


KeyboardInterrupt: 